packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -24,6 +24,12 @@ pub enum ValidationError {
     /// A schema has marked a property with a [`BaseUri`] as required but the [`BaseUri`] does not
     /// exist in the `properties`.
     MissingRequiredProperty(BaseUri),
+    /// Referring to a [`VersionUri`] from a [`BaseUri`] requires that the [`VersionUri::base_uri`]
+    /// matches the [`BaseUri`].
```

Comment:
Perhaps something more like:
```suggestion
    /// When associating a property name with a reference to a Type, we expect the name to match
    /// the [`VersionedUri::base_uri`] inside the reference
```

Or do we validate the base/versioned URIs on things that aren't just property names?

Reply:
I don't know if we also want to check URIs in other locations. Currently, we don't.